src/IB/FEMechanicsExplicitIntegrator.cpp

```diff
@@ -0,0 +1,739 @@
+}
+
+void
+FEMechanicsExplicitIntegrator::forwardEulerStep(const double current_time, const double new_time)
+{
+    const double dt = new_time - current_time;
+    for (unsigned int part = 0; part < d_meshes.size(); ++part)
+    {
+        // U^{n+1} := U^{n} + (dt/rho) F^{n}
+        // X^{n+1} := X^{n} + dt       U^{n}
+        computeLagrangianForce(current_time);
+        int ierr;
+        ierr = VecWAXPY(d_U_vecs->get("new", part).vec(),
+                        dt / d_rhos[part],
+                        d_F_vecs->get("current", part).vec(),
+                        d_U_vecs->get("current", part).vec());
+        IBTK_CHKERRQ(ierr);
+        ierr = VecWAXPY(d_X_vecs->get("new", part).vec(),
+                        dt,
+                        d_U_vecs->get("current", part).vec(),
+                        d_X_vecs->get("current", part).vec());
+        IBTK_CHKERRQ(ierr);
+    }
+    // We copy F^{n} into F^{n+1} to make sure that it is stored in the viz
+    // file.
+    d_F_vecs->copy("current", { "new" });
+}
+
+void
+FEMechanicsExplicitIntegrator::modifiedEulerStep(const double current_time, const double new_time)
+{
+    const double dt = new_time - current_time;
+    for (unsigned int part = 0; part < d_meshes.size(); ++part)
+    {
+        // U^{n+1} := U^{n} + (dt/rho) F^{n}
+        // X^{n+1} := X^{n} + dt       U^{n+1}
+        computeLagrangianForce(current_time);
+        int ierr;
+        ierr = VecWAXPY(d_U_vecs->get("new", part).vec(),
+                        dt / d_rhos[part],
+                        d_F_vecs->get("current", part).vec(),
+                        d_U_vecs->get("current", part).vec());
+        IBTK_CHKERRQ(ierr);
+        ierr = VecWAXPY(d_X_vecs->get("new", part).vec(),
+                        dt,
+                        d_U_vecs->get("new", part).vec(),
+                        d_X_vecs->get("current", part).vec());
+        IBTK_CHKERRQ(ierr);
+    }
+    // We copy F^{n} into F^{n+1} to make sure that it is stored in the viz
+    // file.
+    d_F_vecs->copy("current", { "new" });
+}
+
+void
+FEMechanicsExplicitIntegrator::backwardEulerStep(const double current_time, const double new_time)
+{
+    const double dt = new_time - current_time;
+    for (unsigned int part = 0; part < d_meshes.size(); ++part)
+    {
+        // U^{n+1} := U^{n} + (dt/rho) F^{n+1}
+        // X^{n+1} := X^{n} + dt       U^{n+1}
+        computeLagrangianForce(new_time);
+        int ierr;
+        ierr = VecWAXPY(d_U_vecs->get("new", part).vec(),
+                        dt / d_rhos[part],
+                        d_F_vecs->get("new", part).vec(),
+                        d_U_vecs->get("current", part).vec());
+        IBTK_CHKERRQ(ierr);
+        ierr = VecWAXPY(d_X_vecs->get("new", part).vec(),
+                        dt,
+                        d_U_vecs->get("new", part).vec(),
+                        d_X_vecs->get("current", part).vec());
+        IBTK_CHKERRQ(ierr);
+    }
+}
```

Comment:
Could we move these time step routines to the base class? The only difference between these and the ones in IBFEMethod is the addition of the direct forcing kinematics.

Reply:
No --- `IBFEMethod` does not time step `U`.